### PR TITLE
redirector: restrict mountpoints to /keybase*

### PR DESCRIPTION
**Pull request by strib**
Since this is a suid binary, allowing arbitrary mountpoints is dangerous.

Suggested by oconnor663.

### CODE REVIEW

**Comment by strib** [comment]
> It might be worth being exceptionally conservative and hardcoding the list of keybase/keybase.staging/keybase.devel. We can always change our mind later, right?

Ok done.

> Do you know what happens if the mount target is a symlink? If basil-fuse is willing to traverse a symlink to mount, we might want to be extra conservative and try to prevent it from doing that?

When I try this, I get a nonempty-mount error:

```sh
$ mkdir /tmp/blah
$ sudo ln -s /tmp/blah /keybase.staging
$ KEYBASE_RUN_MODE=staging ./redirector /keybase.staging
2018/03/05 14:01:56 mount helper error: fusermount: mountpoint is not empty
2018/03/05 14:01:56 mount helper error: fusermount: if you are sure this is safe, use the 'nonempty' mount option
Mount error, exiting cleanly: fusermount: exit status 1
``` 

So it seems like it should be safe.

File: redirector/go/main.go [hunk 1]
```diff
@@ -240,6 +240,22 @@ func (r *root) Lookup(
...
+		fmt.Fprintf(os.Stderr, "Usage: %s <mountpoint>\n", os.Args[0])
+		os.Exit(1)
+	}
+
+	// Restrict the mountpoint to paths starting with "/keybase".
+	// Since this is a suid binary, it is dangerous to allow arbitrary
+	// mountpoints.  TODO: Read a redirector mountpoint from a
+	// root-owned config file.
+	r := newRoot()
+	if os.Args[1] != fmt.Sprintf("/%s", r.runmodeStr) {
```
Comment (jzila) [inline redirector/go/main.go]: Maybe this should be case insensitive.

Comment (strib) [reply redirector/go/main.go]: @oconnor663 wanted this to be as restrictive as possible.  We can always change it later if we need   to.

Comment (jzila) [reply redirector/go/main.go]: Understood, thanks.

**Pull request merged**